### PR TITLE
Apply Secrets plugin API

### PR DESCRIPTION
**Pull request by vinokurig**
Signed-off-by: Igor Vinokur <ivinokur@redhat.com>

#### What it does

Corresponding CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23332.
Apply Secrets plugin API: https://code.visualstudio.com/updates/v1_53#_secrets-api

fixes https://github.com/eclipse-theia/theia/issues/9348
#### How to test

1. Clone, compile and start the sample project: https://github.com/vinokurig/secrets-test.
2. Run `Get Password` command and see a notification with empty (undefined) password.
3. Run `Set Password` command and see a notification about password change.
4. Run `Get Password` command and see a notification with the updated password.
5. Run `Delete Password` command and see a notification about password change.
6. Run `Get Password` command and see a notification with empty (undefined) password.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/core/src/node/keytar-server.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        }
+    }
+    async findPassword(service: string): Promise<string | undefined> {
+        const password = await keytar.findPassword(service);
+        if (password) {
+            return password;
+        }
+    }
+    async findCredentials(service: string): Promise<Array<{ account: string, password: string }>> {
+        return [];
```
Comment (azatsarynnyy) [inline packages/core/src/node/keytar-server.ts:91]: maybe remove `findCredentials` or leave a comment explaining why we need to keep it

**Review by azatsarynnyy — APPROVED** [review]
Tested it with the provided extension - works as expected.

File: packages/core/package.json [hunk 2]
```diff
@@ -34,6 +34,7 @@
     "ajv": "^6.5.3",
     "body-parser": "^1.17.2",
     "cookie": "^0.4.0",
+    "keytar": "7.7.0",
```
Comment (vince-fugnitto) [inline packages/core/package.json:37]: I confirmed that the `keytar` dependency is license compatible

File: packages/core/src/browser/credentials-service.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+// code copied and modified from https://github.com/microsoft/vscode/blob/1.55.2/src/vs/workbench/services/credentials/common/credentials.ts#L12
```
Comment (vince-fugnitto) [inline packages/core/src/browser/credentials-service.ts:21]: We likely need a `CQ` for the two files, can you add a link to the corresponding CQ when ready (unless it is already covered)?

```
 https://github.com/microsoft/vscode/blob/1.55.2/src/vs/workbench/services/credentials/common/credentials.ts#L12
https://github.com/microsoft/vscode/blob/1.55.2/src/vs/platform/native/electron-main/nativeHostMainService.ts#L679-L771
```

File: packages/core/src/node/keytar-server.ts (hunk 1 above)
Comment (vinokurig) [reply packages/core/src/node/keytar-server.ts]: Fixed by adding a real implementation

File: packages/core/src/browser/credentials-service.ts (hunk 3 above)
Comment (vinokurig) [reply packages/core/src/browser/credentials-service.ts:21]: Done https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23332

File: packages/core/package.json (hunk 2 above)
Comment (vinokurig) [reply packages/core/package.json:37]: Do I need to create a CQ for this package?

Comment (vince-fugnitto) [reply packages/core/package.json:37]: > Do I need to create a CQ for this package?

@vinokurig I don't think so, it falls under the following [use-case](https://github.com/eclipse-theia/theia/wiki/Registering-CQs#case-new-or-updated-npm-production-dependency). I confirmed that the dependency looks fine, you may double-check if you like following this process.

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto is it OK to merge?

@vinokurig No? The `CQ` is not yet approved.

**Comment by vinokurig** [comment]
@vince-fugnitto The [CQ](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23332) has been approved is there any other objections?

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto The [CQ](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23332) has been approved is there any other objections?

@vinokurig the lat comment I see is:

> This CQ is currently awaiting PMC Approval...

Is it somehow not refreshed?

**Comment by vinokurig** [comment]
@vince-fugnitto 
screenshot-dev eclipse org-2021 05 27-17_17_12
Does this mean that it's approved?

**Comment by vince-fugnitto** [comment]
> @vince-fugnitto Does this mean that it's approved?

@vinokurig I've always waited for there to be an explicit approval comment as part of the CQ, namely:

>  We have completed a high level triage of the attachment and you may now check the content into your project’s source code repository.

**Comment by vinokurig** [comment]
@vince-fugnitto 
> Does this mean that it's approved?

Sorry, forgot to paste my screenshot:
screenshot-dev eclipse org-2021 05 27-17_17_12

>  I've always waited for there to be an explicit approval comment as part of the CQ, namely:
>
>>We have completed a high level triage of the attachment and you may now check the content into your project’s source code repository.

OK, let's wait for the comment.

**Comment by vinokurig** [comment]
@vince-fugnitto since the [CQ](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23332) has been approved I am going to merge it tomorrow, if now other objections?

**Pull request merged**